firmware/src/hal/teensy/hal_teensy.cpp

```
@@ -74,6 +74,16 @@ void start_timer(uint32_t period_us, TimerCallback callback) {
 void lock() { noInterrupts(); }
 void unlock() { interrupts(); }
 
+// The MIDI wire and the sync jack, still undriven: Serial1 and the pins WIRING.md
+// reserves are wired up in link_teensy.cpp, which arrives with the commit that
+// puts a scope on them. Until then the device says it has no port, which is the
+// truth about this firmware and not about the hardware.
+int read_clock_in(ClockIn*, int) { return 0; }
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)